core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -172,6 +184,21 @@ private boolean addMember(Address joiner, CacheJoinInfo joinInfo) {
             return false;
          }
 
+         // Validate if the member is allowed to join
+         if (persistentState.isPresent()) {
```

Comment:
I'm just worried that if the merge coordinator doesn't know the persistent uuids, it doesn't have any way to signal that to the nodes that already sent the join request to the old coordinator, so they'll time out instead of reporting an error immediately.